android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt

```diff
@@ -20,58 +23,88 @@ class RNPianoAnalyticsModule internal constructor(context: ReactApplicationConte
   // CONFIG
   @ReactMethod
   override fun setConfiguration(collectDomain: String, siteId: Int) {
-    var config = Configuration.Builder()
-      .withCollectDomain(collectDomain)
-      .withSite(siteId)
+    var config = Configuration.Builder(collectDomain = collectDomain,
+      site = siteId)
       .build()
-    piano.setConfiguration(config)
+    piano.init(context, config)
   }
 
   // EVENTS METHODS
   @ReactMethod
   override fun sendEvent(eventName: String, params: ReadableMap) {
-    piano.sendEvent(Event(eventName, params.toHashMap() as Map<String, String>))
+
+  // TODO : => il faut remplacer cette fonction par une boucle HashMap =>  old Method : piano.sendEvent(Event(eventName, params.toHashMap() as Map<String, String>))
+    PianoAnalytics.getInstance().sendEvents(
+      Event.Builder(eventName)
+        .properties(
+          Property(PropertyName("page"), "Page name"),
+          Property(PropertyName("page_premium"), false),
+          Property(PropertyName("page_authors"), arrayOf("Clark Kent","John Cena")),
+          Property(PropertyName("last_update"), 1698225000, Property.Type.DATE),
+          // you can add properties here
+        )
+        .build()
+    )
+
   }
 
   // USER METHODS
   @ReactMethod
   override fun setUser(id: String, category: String?, enableStorage: Boolean) {
-    piano.setUser(id, category, enableStorage)
+    val userStorage = PianoAnalytics.getInstance().userStorage
+    // set current user
+    userStorage.currentUser = User(id, category, enableStorage)
+  //  old Method : piano.setUser(id, category, enableStorage)
   }
 
   @ReactMethod
   override fun deleteUser() {
-    piano.deleteUser()
+    PianoAnalytics.getInstance()
+//    TODO : add function DELETE USER   sur la doc officiel il ny  pas de methode pour la version 3.3.0
+//  old Method :    piano.deleteUser()
   }
 
   // PRIVACY MODE
   @ReactMethod
   override fun privacySetMode(mode: String) {
-    piano.privacySetMode(mode)
+    PianoAnalytics.getInstance().privacyModesStorage
+// DEPRECATED ?  utiliser consent a la place ?
+//    old Method : piano.privacySetMode(mode)
```

Comment:
check this for privacyMode :
![Uploading Capture d’écran 2024-06-12 à 15.39.04.png…]()